Configurations/15-android.conf

```diff
@@ -22,6 +24,7 @@
 
             my $ndk = $ENV{ANDROID_NDK};
             die "\$ANDROID_NDK is not defined"  if (!$ndk);
+            $ndk = canonpath($ndk);
             die "\$ANDROID_NDK=$ndk is invalid" if (!-d "$ndk/platforms");
```

Comment:
There was suggestion to word the message as "\$ANDROID_NDK=$ENV{ANDROID_NDK} is invalid" so that user observes the actual, non-"combed" value set by user. Even if user is not observant, *we* would be interested to see the actual value in case problem report is filed.

Reply:
Will it matter much?  `-d` should work correctly with uncanonised paths as well...  in other words, if `ANDROID_NDK=/foo//bar/cookie/` and `/foo/bar/cookie/platforms` exist, canonisation of the path won't matter here.  Put in other words if `-d` fails, it won't be because of redundant slashes!

Reply:
As already said, it would be *more pedagogical*. This has no bearing to how -d works, so of course it doesn't matter much in respect to useful work script performs. It's about how it interacts with user in case no useful work can be done.

Reply:
I'm not sure which is most pedagogical, to be honest, but here ya go: #6925